src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+= Running Hawkular Agent
+John Mazzitelli
+2017-12-11
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, agent, prometheus, metrics
+
+Now that Hawkular is moving towards integraton with Prometheus as its metrics collection and storage system, the agent
+has had some changes. This document will discuss those changes and how to run the Hawkular Agent.
+
+First, the Hawkular Agent now only runs as a javaagent (no longer does it run inside WildFly as a subsystem extension).
+So to run the agent, simply attach it as a javaagent to whatever JVM you want to monitor (be it a WildFly or EAP app server or
+any JVM application that exposes metrics via JMX). For example, add this command line option to the command that starts the JVM:
+
+```
+-javaagent:hawkular-javaagent.jar=config=hawkular-javaagent-config.yaml,delay=10
+```
+
+The configuration file contains all the settings necessary to start the agent and connect to the Hawkular Server.
+Once connected, the agent will pull down additional configuration settings from the server.
+
+=== Running in WildFly
+
+If you are attaching the agent to a WildFly server, you must set some additional settings.
+
+If you are running WildFly in standalone mode, the standalone.conf you should have something like:
```

Comment:
Question - Is this the default configuration?  Or an example of what the user wants their configuration to look like?  If the latter, this might be better presented as a table with the configuration options, followed by an example of a properly configured standalone.conf file.

Reply:
What we want the configuration to look like.

Remember, we have no control of the customer's wildfly installation - they must add some additional JVM command line options when starting their java app to get the javaagent to attach to it.

In this case, this is not "the default" configuration because hawkular agent is nowhere in base, default WildFly and we don't have an installer.

These lines describe how a user can modify their WildFly's domain.conf/standalone.conf file (which are nothing more than unix script files). So I'm not sure a table is adequate - we should tell them what the lines of script should/could look like.